source/unified-test-format/unified-test-format.rst

```diff
@@ -273,6 +274,21 @@ Test runners MUST support the following types of entities:
   not be supported by all drivers and could yield runtime errors (e.g. while
   loading a test file with an Extended JSON parser).
 
+- Test runner thread. An entity representing a "thread" that can be used to
+  concurrently execute operations. Thread entities MUST be able to run
+  concurrently with the main test runner thread and other thread entities, but
+  they do not have to be implemented as actual OS threads (e.g. they can be
+  goroutines or async tasks). See `entity_thread`_ for more information on how
+  they are created.
+
+.. _entity_topologydescription:
+
+- TopologyDescription. An entity representing a client's `TopologyDescription
```

Comment:
In the SDAM integration test format, there was an operation called `recordPrimary` that was used exclusively for a subsequent `waitForPrimaryChange` operation. I figured that it would be best to generalize this to be more widely useful and leverage the entity system, so I opted for treating the whole TopologyDescription as an entity. The nice thing about this is it lets us perform arbitrary assertions on the state of the topology in the tests now. For instance, I also added the `assertTopologyType` operation to ensure that the primary has indeed been found in the SDAM test that uses this.

Let me know if this approach makes sense.